packages/ckeditor5-dev-release-tools/lib/tasks/commitandtag.js

```diff
@@ -39,6 +51,33 @@ export default async function commitAndTag( { version, files, cwd = process.cwd(
 		return;
 	}
 
-	await git.commit( `Release: v${ version }.${ skipCi ? ' [skip ci]' : '' }`, filePathsToAdd );
-	await git.addTag( `v${ version }` );
+	if ( dryRun ) {
+		await dryRunGitCommit( filePathsToAdd, preCommitCommand, cwd );
+		listrTask.output = `[dry run] Creating git tag v${ tagForVersion }.`;
+	} else {
+		await git.commit( `Release: v${ version }.${ skipCi ? ' [skip ci]' : '' }`, filePathsToAdd );
+		await git.addTag( `v${ version }` );
+	}
+}
+
+/**
+ * Function that imitates running dry run for the `git commit` command.
+ *
+ * @param {Array.<string>} filePathsToAdd File paths to run the dry run validations for.
+ * @param {string} preCommitCommand Command to run before creating a commit for the dry run mode.
+ * @param {string} cwd Current working directory from which all paths will be resolved.
+ * @returns {Promise<void>}
+ */
+async function dryRunGitCommit( filePathsToAdd, preCommitCommand, cwd ) {
```

Comment:
```suggestion
async function executePreCommitHook( { cwd, git, hookCommand, filesToCommit } ) {
```
I propose something like this. I am unsure if we can get `cwd` from the `git` instance. I checked docs, but it allows changing it only.

Reply:
It looks like it's not possible to get `cwd` from the `simple-git` instance.